sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -56,27 +57,13 @@ public void Bar(bool inParameter, out bool outParameter)
         [TestCategory("Symbolic execution")]
         public void ExplodedGraph_SequentialInput()
         {
-            var testInput = "var a = true; var b = false; b = !b; a = (b);";
-            var method = ControlFlowGraphTest.CompileWithMethodBody(string.Format(TestInput, testInput), "Bar", out var semanticModel);
-            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
-            var aSymbol = GetSymbol(semanticModel, method, "a");
-            var bSymbol = GetSymbol(semanticModel, method, "b");
-
-            var cfg = CSharpControlFlowGraph.Create(method.Body, semanticModel);
-            var lva = CSharpLiveVariableAnalysis.Analyze(cfg, methodSymbol, semanticModel);
-
-            var explodedGraph = new CSharpExplodedGraph(cfg, methodSymbol, semanticModel, lva);
-            var explorationEnded = false;
-            explodedGraph.ExplorationEnded += (sender, args) => { explorationEnded = true; };
-
-            var numberOfExitBlockReached = 0;
-            explodedGraph.ExitBlockReached += (sender, args) => { numberOfExitBlockReached++; };
-
-            var numberOfProcessedInstructions = 0;
-            explodedGraph.InstructionProcessed +=
+            const string testInput = "var a = true; var b = false; b = !b; a = (b);";
+            var context = new ExplodedGraphContext(testInput);
+            var aSymbol = context.GetSymbol("a");
+            var bSymbol = context.GetSymbol("b");
+            context.EG.InstructionProcessed +=
```

Comment:
I would prefer to keep `ExplodedGraph` instead of `EG` for readability inside the context class. WDYT ?